src/main/java/oracle/r2dbc/impl/UserDefinedType.java

```diff
@@ -0,0 +1,23 @@
+package oracle.r2dbc.impl;
+
+import io.r2dbc.spi.Type;
+
+public final class UserDefinedType implements Type {
+    private final Class sqlClassType;
```

Comment:
I had been thinking to add new Type constants in OracleR2dbcTypes:
STRUCT, ARRAY, and REF.

And these SQL types can map to Java classes of the java.sql package: Struct, Array, and Ref

All this is to say that UserDefinedType can retain an io.r2dbc.spi.Type field, rather than a Class field. The Type field would be one of STRUCT, ARRAY, REF. We can then implement getJavaType to return the java type mapping for this Type field.

Does this make sense?